controllers/controllers.go

```diff
@@ -17,20 +22,34 @@ type Routes struct {
 	Public  *gin.RouterGroup
 }
 
+var tokenCache *cache.Cache
+
 func AuthenticationRequired(auths ...string) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		// TODO
+		id_token := c.GetHeader("ID-Token")
+		_, err := authenticate(id_token)
+		if err != nil {
+			log.Println("could not authenticate id token -- " + err.Error())
+			c.JSON(http.StatusUnauthorized, gin.H{"error": "could not authenticate id token -- " + err.Error()})
+			c.Abort()
+			return
+		}
+		// add token to cache, set it to expire after an hour (I'm basically just using this as a list)
+		// should I be using their id or their id_token? Does it make a difference?
+		tokenCache.Add(id_token, "dummy", time.Hour)
```

Comment:
I believe you should be using the idToken, since we don't have access to the id without asking Google's API first, which is what we are trying to avoid when we cache.
Also, does this function reset the timer on a key if it already exists in the cache?